src/Schema/AbstractAsset.php

```diff
@@ -28,11 +33,22 @@
  * This encapsulation hack is necessary to keep a consistent state of the database schema. Say we have a list of tables
  * array($tableName => Table($tableName)); if you want to rename the table, you have to make sure this does not get
  * recreated during schema migration.
+ *
+ * @internal This class should be extended only by DBAL itself.
+ *
+ * @template N of Name
  */
 abstract class AbstractAsset
 {
     protected string $_name = '';
 
+    /**
+     * Indicates whether the object name has been initialized.
+     *
+     * @internal
```

Comment:
This seems redundant. If the class is internal, then aren't all of its properties internal?

Reply:
It is redundant. I decided to mark the whole class as internal last minute.